.github/workflows/ci.yml

```diff
@@ -1,6 +1,11 @@
 name: CI
 
-on: [push, pull_request, workflow_dispatch]
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+  workflow_dispatch:
```

Comment:
Because in the way we have it till now the CI gets "double triggered" because of PR and "push". See: https://discord.com/channels/895609329053474826/963349623714504784/1301528920893558855